README.md

```diff
@@ -113,6 +113,9 @@ collector_opts:
     blacklist:
       - docker0
       - lo
+logging_levels:
+  root: INFO
+  foomodule.barcollector: WARNING
```

Comment:
This is not that optimal. Configuration for should be placed in `collector_opts.<collector_name>`. That leads to a structure sth like this:

```
collector_opts:
  example:
    log_level: WARNING
``` 

Can you please take this change in consideration?

Reply:
I agree that configuring the logging levels in the `collector_opts` for each collector would be semantically cleaner. But that would require changes to all collector-implementations out there: because there is no correlation between the name of a logger / channel and the  name of the python-package that uses it, this couldn't be implemented by `p3exporter`. In fact, in my use-case I am configuring not the logging of the actual collector but of a package used by that collector...

The requested change would require all classes derived from `CollectorBase` to parse this extra entry in the `CollectorConfig.collector_opts`. And that is assuming that there is no collector out there that already implemented this setting but with a different semantic, e.g. as a dict of sub-modules...

I am open to suggestions how to make the requested change locally in `p3exporter` if I am overlooking something.

Reply:
The change would only affect the `CollectorBase` class and such, it can provide methods for managing logging in each collector.

Reply:
You mean like adding a method `CollectorBase.setLoggers(logger_names)` that the deriving collectors call from their constructors? Maybe add an optional argument to `CollectorBase.__init__()` as a shortcut? I can do that (please indicate if I understood you correctly and I will start on that).

But as this depends on the collectors being updated to use this. I would therefore like to keep the global `logging_levels` config in addition to the above per-collector one.

Reply:
> please indicate if I understood you correctly and I will start on that

`CollectorBase.setLoggers(logger_names)` is what I thought of. Thank you for implementing. It would be great if you also show how to apply the new method in the distributed collectors.

> I would therefore like to keep the global `logging_levels` config in addition

Okay, but I would have some adjustments to be more flexible. Starting from the configuration, I would prefer a structure like this:

```
logging:
  - name: <class>:
    level: <LEVEL>
    target: <target file, default to STDOUT>
```

With this, the structure of your proposal would look as followed:

```
logging:
  - name root
    level: INFO
  - name: foomodule.barcollector: 
    level: WARNING
    target: /path/to/my/collector/logfile.log
```

Can you deal with that extension to your idea?

Reply:
Will take a bit if time, but will do so.

Reply:
Done.

Reply:
Thank you for your effort. Implementing the requested changes in under 6 hours is really a good pace. 

May I also ask for your opinion? Do you mean that `logging[].name` is a good choice, or should it better named `logging[].class_name`?

Reply:
I think `name` is a good choice. And specifically compared to `class_name`: these are arbitrary names given to loggers, I could call one "Egg, bacon, sausage, and Spam" if I really wanted to. The only special character in there is the dot (`.`) which is used to determine the hierarchy of loggers, AFAICT.

The only contenders from my POV would be `logger_name` and `channel_name`, but I think `name` is still preferable here.